framework/PostProcessors/ParetoFrontierPostProcessor.py

```diff
@@ -0,0 +1,147 @@
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on March 25, 2020
+
+@author: mandd
+"""
+from __future__ import division, print_function , unicode_literals, absolute_import
+
+#External Modules---------------------------------------------------------------
+import numpy as np
+#External Modules End-----------------------------------------------------------
+
+#Internal Modules---------------------------------------------------------------
+from .PostProcessor import PostProcessor
+from utils import utils
+from utils import InputData, InputTypes
+import Runners
+#Internal Modules End-----------------------------------------------------------
+
+class ParetoFrontier(PostProcessor):
+  """
+    This postprocessor selects the points that lie on the Pareto frontier
+    The postprocessor acts only on PointSet and return a subset of such PointSet
+  """
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    inputSpecification = super(ParetoFrontier, cls).getInputSpecification()
+    inputSpecification.addSub(InputData.parameterInputFactory('costID' , contentType=InputTypes.StringType))
+    inputSpecification.addSub(InputData.parameterInputFactory('valueID', contentType=InputTypes.StringType))
+    return inputSpecification
+
+  def initialize(self, runInfo, inputs, initDict) :
+    """
+      Method to initialize the PostProcessor
+      @ In, runInfo, dict, dictionary of run info (e.g. working dir, etc)
+      @ In, inputs, list, list of inputs
+      @ In, initDict, dict, dictionary with initialization options
+      @ Out, None
+    """
+    PostProcessor.initialize(self, runInfo, inputs, initDict)
+
+  def _localReadMoreXML(self, xmlNode):
+    """
+      Function to read the portion of the xml input that belongs to this specialized class
+      and initialize some stuff based on the inputs got
+      @ In, xmlNode, xml.etree.Element, Xml element node
+      @ Out, None
+    """
+    paramInput = ParetoFrontier.getInputSpecification()()
+    paramInput.parseNode(xmlNode)
+    self._handleInput(paramInput)
+
+  def _handleInput(self, paramInput):
+    """
+      Function to handle the parsed paramInput for this class.
+      @ In, paramInput, ParameterInput, the already-parsed input.
+      @ Out, None
+    """
+    costID  = paramInput.findFirst('costID')
+    self.costID  = costID.value
+    valueID = paramInput.findFirst('valueID')
+    self.valueID = valueID.value
```

Comment:
FYI, currently, the InputData can not check if the subnode is required or not.

Reply:
As far as I Know, we do not check it now, and we plan to do it in the near future.